master/pkg/schemas/defaults.go

```diff
@@ -45,52 +45,55 @@ func getDefaultSource(obj reflect.Value) interface{} {
 	return nil
 }
 
+func allocateWithDefaultBytes(typ reflect.Type, defaultBytes []byte) reflect.Value {
```

Comment:
I'm beginning to think this should all return errors not panic. Yes, a panic means bad code, but that means bad code in a bad spot (not in an actor or http handler, which is basically none of our code but still) is unrecoverable which kinda freaks me out.

Reply:
I mean, you've been saying that for a while.

I still don't think so though.  These errors can only occur with invalid default values, which are hardcoded json values and should be tested beforehand to be valid.

A better language would execute them at compile time as some sort of const expression.  In this case we can compensate for go with a bunch of tests.

But overall, something that conceptually means "take this struct and fill nil pointers with constant values" should not throw an error for anything other than a malloc failure (in which case Go panics anyway).

I could write a test that checks every default to make sure it is valid json if you'd like?

Reply:
Nah that's way too much work. We should just be really careful to write tests for every expconf thing?